client/src/Components/Results.js

```diff
@@ -1,24 +1,143 @@
+			return("corect");
+		} else{
+			return("incorect");
+		}
+	};
+	const findStudenAnswerResult=(actualQuestionId, oneName)=>{
+		let studentAnsweredId = allResults.filter((el)=>el.question_id==actualQuestionId);
+		let studentAnsweredQuestion = studentAnsweredId.filter((el)=>el.studentName==oneName);
+		if(studentAnsweredQuestion.length==0){
+			return("unknown");
+		}
+		if(studentAnsweredQuestion.length>0){
+		}
+		if(studentAnsweredQuestion!==undefined&&studentAnsweredQuestion[attemptNumber]){
+			return(studentAnsweredQuestion[attemptNumber].value);
+		}
+	};
+	useEffect(() => {
+		fetchResults();
+	},[quizRoute]);
+	useEffect(() => {
+		if(quizSelected!==null){
+			const makeQuestions=()=>{
+				let selectedQuizQuestions= [];
+				selectedQuizQuestions = quizSelected.questions_id.map((selId)=>{
+					let found = (props.questions.find((question)=>question._id==selId));
+					selectedQuizQuestions.push(found);
+					setSelectedQuizQuestions(selectedQuizQuestions);
+				}
+				);
+			};
+			makeQuestions();
+		}
+	},[quizSelected]);
+	const makeNames=()=>{
+		let tempNames = [];
+		 if(allResults!=="not found!"){
+			allResults.map((oneAnswer)=>{
+				if(studentNames&&!tempNames.includes(oneAnswer.studentName)){
+					tempNames.push(oneAnswer.studentName);
+				}
+			});
+			setStudentNames(tempNames);
+		}
+	};
+	useEffect(() => {
+		if(allResults){
+			makeNames();
+		}
+	},[allResults]);
 
+	const quizzToSeeResultsChosen=(e)=>{
+		setQuizRoute(e.target.value);
+		const selectedQuiz=props.quizes.find((quiz)=>(quiz._id==e.target.value));
+		setSelectedQuizQuestions([]);
+		setQuizSelected(selectedQuiz);
+	};
+	if(allResults=="not found!"){
+		return(
+			<div className='col-12 centered'>
+				<h1> no results for this quiz have been submitted yet</h1>
+				<div className='col-12'>
+					<h1>Welcome to QuizzTime</h1>
+					<select name="quizzez" className="quizzez"  onChange={quizzToSeeResultsChosen}>
+						<option >select a quiz</option>
+						{props.quizes.map((quiz)=>{
+							return(<option value={quiz._id} >{quiz.name}</option>);
+						})}
+		  </select>
+				</div></div>);
+	} else if(props.quizes){
+		return (
+			<div  className='col-12 centered'>
+				<h1>Chosse a quizz to see student results</h1>
+				<select name="quizzez" className="quizzez" onChange={quizzToSeeResultsChosen}>
+					<option value="" disabled selected hidden>select a quiz</option>
 					{props.quizes.map((quiz)=>{
-						return(<option value={quiz.name}>{quiz.name}</option>);
+						return(<option value={quiz._id} >{quiz.name}</option>);
 					})}
+		  </select>
 
-		  </select> */}
-		  </div>
-		);
+				{quizSelected&&allResults&&allResults!=="not found!"?<div><div className='centered'><h3>attempt number {attemptNumber+1} </h3><p>(change to see if some students made multiple attempts)</p></div>
+					<button onClick={decreaseAttemptNumber}>decrease attempt number </button>
+					<button onClick={increaseAttemptNumber}>increase attempt number </button><table>
```

Comment:
here you don't need to call functions, instead of decrease just use: setAttemptNumber(attemptNumber - 1)
it's work perfectly